### PR TITLE
Auth for the public ALB's HTTPS listener

### PR DESCRIPTION
**Pull request by thisismana**
User can activate COGNITO authentication by passing:
```
module "service" {
  # ...
  alb_cogino_pool_arn           = data.terraform_remote_state.cognito.outputs.alb_cogino_pool_arn
  alb_cogino_pool_client_id     = data.terraform_remote_state.cognito.outputs.alb_cogino_pool_client_id
  alb_cogino_pool_domain        = data.terraform_remote_state.cognito.outputs.alb_cogino_pool_domain
}
```

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -47,6 +47,24 @@ variable "alb_listener_priority" {
   type        = number
 }
 
+variable "alb_cogino_pool_arn" {
+  type        = string
+  default     = null
```
Comment (moritzzimmer) [inline variables.tf:52]: we could also used empty strings as defaults

File: alb.tf [hunk 2]
```diff
@@ -48,6 +48,18 @@ resource "aws_alb_listener_rule" "public" {
   listener_arn = data.aws_lb_listener.public[count.index].arn
   priority     = var.alb_listener_priority
 
+  dynamic "action" {
+    for_each = var.alb_cogino_pool_arn == null ? [/*noop*/] : ["enabled"]
```
Comment (moritzzimmer) [inline alb.tf:52]: we could also check for empty string instead of `null`, but fine for me

File: variables.tf (hunk 1 above)
Comment (thisismana) [reply variables.tf]: done

File: alb.tf (hunk 2 above)
Comment (thisismana) [reply alb.tf]: done.

**Pull request merged**